bindings/nodejs/lib/types/unlock_condition.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { Address, AliasAddress } from './address';
+
+/**
+ * All of the unlock condition types.
+ */
+enum UnlockConditionType {
+    Address = 0,
+    StorageDepositReturn = 1,
+    Timelock = 2,
+    Expiration = 3,
+    StateControllerAddress = 4,
+    GovernorAddress = 5,
+    ImmutableAliasAddress = 6,
+}
+
+abstract class UnlockCondition {
+    private type: UnlockConditionType;
+
+    constructor(type: UnlockConditionType) {
+        this.type = type;
+    }
+    /**
+     * The type of unlock condition.
+     */
+    getType(): UnlockConditionType {
+        return this.type;
+    }
+}
+
+class AddressUnlockCondition extends UnlockCondition /*implements IAddressUnlockCondition*/ {
+    private address: Address;
+    constructor(address: Address) {
+        super(UnlockConditionType.Address);
+        this.address = address;
+    }
+
+    /**
+     * The address.
+     */
+    getAddress(): Address {
+        return this.address;
+    }
+}
+/**
+ * Storage Desposit Return Unlock Condition.
+ */
+class StorageDepositReturnUnlockCondition extends UnlockCondition /*implements IStorageDepositReturnUnlockCondition*/ {
+    private amount: string;
+    private returnAddress: Address;
+
+    constructor(returnAddress: Address, amount: string) {
+        super(UnlockConditionType.StorageDepositReturn);
+        this.amount = amount;
+        this.returnAddress = returnAddress;
+    }
+    /**
+     * Amount of IOTA tokens the consuming transaction should deposit to the address defined in return address.
+     */
+    getAmount(): string {
+        return this.amount;
+    }
+
+    /**
+     * The return address.
+     */
+    getReturnAddress(): Address {
+        return this.returnAddress;
+    }
+}
+/**
+ * Timelock Unlock Condition.
+ */
+class TimelockUnlockCondition extends UnlockCondition /*implements ITimelockUnlockCondition*/ {
+    private unixTime: number;
+
+    constructor(unixTime: number) {
+        super(UnlockConditionType.Timelock);
+        this.unixTime = unixTime;
+    }
+    /**
+     * Unix time (seconds since Unix epoch) starting from which the output can be consumed.
+     */
+    getUnixTime(): number {
+        return this.unixTime;
+    }
+}
+
+class ExpirationUnlockCondition extends UnlockCondition /*implements IExpirationUnlockCondition*/ {
+    private returnAddress: Address;
+    private unixTime: number;
+
+    constructor(returnAddress: Address, unixTime: number) {
+        super(UnlockConditionType.Expiration);
+        this.returnAddress = returnAddress;
+        this.unixTime = unixTime;
+    }
+    /**
+     * Unix time (seconds since Unix epoch) starting from which the output can be consumed.
```

Comment:
Wrong comment, can be unlocked before expired as well (by address unlock conditions, later by return address)